src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java

```diff
@@ -109,12 +121,14 @@ public static LocalDateTime parseTimedTaskTime(String[] nameKeywords, int length
         int timePrefixLocation = -1;
 
         for (int i = 2; i < length; i++) {
-            if (nameKeywords[i].substring(0, 2).equals(TASK_DATE)) {
+            if (nameKeywords[i].length() <= 1) {
+                //Ignores the string segment if the length is <= 1
+
+            } else if (nameKeywords[i].substring(0, 2).equals(TASK_DATE)) {
```

Comment:
same comment as above on shortening code

Reply:
same comment as the above